### ISSUE TITLE
Import of DejaGnu file does not finish (soon enough) and is possibly incomplete

### ISSUE THREAD

**Issue by petr-muller** [FilipMatys/IBP2013#51]
I periodically import SystemTap testsuite results into Corly. It seems that the import of these results never end in a sufficient time: I have a timeout set to 30m. This seems to happen with both manual import in Web UI and using the import script.

Testcase: http://www.afri.cz/files/systemtap-2015-08-27.sum

Additionally, it seems this file is not properly imported. After import I see this in the [summary](http://hele.fit.vutbr.cz/~xmatys10/production/#/home/project/1/submission/49):
UNTESTED	92
PASS	5664
FAIL	50
KFAIL	77
XFAIL	320
XPASS	1
UNSUPPORTED	7

But:
```
$ grep "^UNTESTED:" 2015-08-27.sum  | wc -l
100
$ grep "^PASS:" 2015-08-27.sum  | wc -l
6440
$ grep "^FAIL:" 2015-08-27.sum  | wc -l
50
$ grep "^KFAIL:" 2015-08-27.sum  | wc -l
89
$ grep "^XPASS:" 2015-08-27.sum  | wc -l
1
$ grep "^UNSUPPORTED:" 2015-08-27.sum  | wc -l
7
```

**Comment by petr-muller** [comment]
I have performed an additional test: In one browser, I tried to import the testcase linked above. In the second browser, I have tried to look at the submission page. The submission page was created immediately, and when refreshing, the progress during import was clearly visible (the numbers were rising).

After cca 5 minutes, the numbers stopped rising and stabilized (again) on the numbers summary in the initial comment. The browser and tab where I run the import was still running, though.

**Comment by FilipMatys** [comment]
The server execution time is limited.

Tried parallel processing, but it comes to the same problem thanks to the need of waiting for threads to finish. More complicated threading leads to application failure (too many threads thanks to huge amount of test cases).

Right now, reseting the timeout when executing database operation seems to work. Managed to import the whole file on local machine, where I currently have execution time limit set to 30 seconds.

Will test on the server and then, if it will work, deploy to production.

**Comment by bond95** [comment]
Can we use for parsing results other scripts, for example on python, it will work faster than on php?

**Comment by petr-muller** [comment]
Importing http://www.afri.cz/files/2015-09-11.sum into Test instance via WebUI: The "Uploading" wheel keeps spinning for several hours, never ends. The record http://www.result-cloud.org/test/#/home/project/1/submission/20 is created with 5694 PASS, 51 FAIL etc. According to the summary in the file itself, there should be 5694 PASS results.

**Comment by petr-muller** [comment]
Importing http://www.afri.cz/files/2015-09-11.sum into Production instance via WebUI: The "Uploading" wheel keeps spinning for several hours, never ends. The record http://result-cloud.org/production/#/home/project/6/submission/110 is created with 5694 PASS, 51 FAIL etc. According to the summary in the file itself, there should be 5694 PASS results.

**Comment by petr-muller** [comment]
Importing http://www.afri.cz/files/2015-09-19.sum into Production instance via script: The import was killed by a 30min timeout. The record http://result-cloud.org/production/#/home/project/1/submission/111 with 5694 PASS, 51 FAIL etc. According to the summary in the file itself, there should be 6482 PASS results.

**Comment by FilipMatys** [comment]
Found a problem in the systemtap parser, that caused the last test case to be thrown away. This problem is now fixed. 

Did some basic analysis:
Parser finished in 0.10488986968994 seconds
Saving finished in 230.90884399414 seconds
Import finished in 491.38591003418 seconds (Overall with dashboard)

So the most time consuming part is the core with save operations and dashboard creation.

**Comment by FilipMatys** [comment]
I managed to achieve these results:

Parser finished in 0.10406708717346 seconds
Saving finished in 2.9170730113983 seconds
Import finished in 8.7618288993835 seconds

Solution? Changed database engine.... Will deploy the changes on production and test. The fix for the parser will be available later. (Dont have access to "hele" computer).

I also tried to insert data with Phalcon framework, but its results werent much better.
